src/v/pandaproxy/json/iobuf.h

```diff
@@ -78,12 +80,23 @@ class rjson_serialize_impl<iobuf> {
     bool
     encode_base64(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
         if (buf.empty()) {
-            return w.String("", 0);
+            return w.Null();
         }
         // TODO Ben: Implement custom OutputStream to prevent this linearization
         return w.String(iobuf_to_base64(buf));
     };
 
+    bool encode_json(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        if (buf.empty()) {
+            return w.Null();
+        }
+        iobuf_parser p{std::move(buf)};
+        auto str = p.read_string(p.bytes_left());
+        rapidjson::Reader reader;
+        rapidjson::StringStream ss{str.c_str()};
```

Comment:
I think this is an issue no? what if there is a null `\0` in the string. why not pass the str.size() too ? 
how does it know when to end ?

Reply:
Does JSON support embedded `\0`?
I can't pass `size()` because `StringStream` doesn't accept `size()`.
It ends when it hits `\0`.

Reply:
a comment would be helpful there.

Reply:
Even better - you get a `static_assert`.